### PR TITLE
Hide economy setting when default target quantity is 0

### PR DESCRIPTION
**Pull request by hessenfarmer**
…e eco setting to be hidden

used for atlantean blackrootflour so far.

### CODE REVIEW

File: data/tribes/wares/blackroot_flour/init.lua [hunk 1]
```diff
@@ -7,7 +7,10 @@ tribes:new_ware_type {
    descname = pgettext("ware", "Blackroot Flour"),
    helptext_script = dirname .. "helptexts.lua",
    icon = dirname .. "menu.png",
-   default_target_quantity = {},
+   default_target_quantity = {
+      atlanteans = 0
+
```
Comment (Noordfrees) [inline data/tribes/wares/blackroot_flour/init.lua]: remove blank line

**Review by Noordfrees — COMMENTED** [review]
Code LGTM, one style comment.

I tested this and found:
- If I set the target for cornflour to a very high value, then blackroot flour will only be produced if there is demand for it. Otherwise, never.
- If I set the cornflour target to a low value, the mill produces blackroot flour indefinitely.

Only downside I can think of is that it is impossible to stockpile blackroot flour even when cornflour is needed, but that's not really a problem IMHO.

Please add a hint in the documentation that wares with a default target quantity of 0 will be excluded from the economy target settings window.

File: data/tribes/wares/armor/init.lua [hunk 2]
```diff
@@ -39,6 +39,10 @@ dirname = path.dirname(__file__)
 --
 --    **default_target_quantity**: A table listing the default target quantity
 --    for each tribe's economy. For example, ``{ atlanteans = 3, empire = 1 }``
+--    This parameter is optional. If not set the economy will always demand this ware.
```
Comment (Noordfrees) [inline data/tribes/wares/armor/init.lua]: That's not correct. The table always has to be specified, but it may be empty.

Comment (hessenfarmer) [reply data/tribes/wares/armor/init.lua]: oops you are right

**Comment by Noordfrees** [comment]
Good job :)

**Pull request merged**